README.md

```diff
@@ -9,9 +9,29 @@ Also included:
   + APIs for packet data over radio [README_data](README_data.md)
   + An STM32 embedded version of FreeDV 1600/700D/700E for the [SM1000](stm32/README.md)
 
-## Older code
+## Old Code and Deprecated Features
 
-In July 2023 this repo was refactored, older code can be found in https://github.com/drowe67/codec2-dev
+In July 2023 this repo was refactored, older code can be found in https://github.com/drowe67/codec2-dev. 
+
+We are currently conducting a major re-development of Codec 2, new speech coding, modems, and FreeDV modes are under active development.  We have limited resources available for maintenance or development on features that may soon be replaced (with the exception of major bugs).  We'd rather put our efforts into new features! Modes and Features we are not actively maintaining at present, and likely to be superseded in the near future include:
+1. FreeDV 1600, 700C, 700D, 700E, 2020.
+1. Codec 2 modes with the exception of Codec 2 3200 for M17.
+1. Our fork of LPCNet.
+1. The stm32/SM1000 firmware (paused until new modes available).
+
+The code supporting these modes won't be going away any time soon (and we will continue to include any modes/code in popular use), but we have chosen not to actively develop it at this time.
+
+## Pull Requests, Feature Requests
+
+We have a process for considering Feature Requests and Pull Requests that we will guide you through.
+
+Feature Requests can be submitted via GitHub Issues.
+
+Before writing any code or submitting a PR - **please discuss** the PR with developers by raising a GitHub Issue. We have many years of experience and a carefully considered plan for Codec 2 development, and can guide you on work that will most benefit this project.
+
+## Ports to non gcc/C99 Compilers
+
+We have standardized on gcc/C99.  We and encourage people who want to use non-standard compilers like MSVC and certain embedded compilers to maintain their own Codec 2 forks. Our focus needs to be on what’s unique about our project – the speech codec and modem waveforms – rather than consuming time and resources on non-core activities that others can do equally well. If you decide to fork Codec 2 to a non C99/gcc compiler - please ensure you port the ctests and that they all pass.  If the tests have not been ported or do not pass - it's not Codec 2.
```

Comment:
Should Clang get mentioned as well, or does that fall under gcc?

Reply:
Not sure ...  do we support/do any compiling with clang?  We should probably make it specific, so we know what we should/shouldn't support.  We have agreed to use clang for code formatting.

Reply:
Clang is used for macOS and Windows binaries. Not sure what Linux distros use it by default if any.

Reply:
Cool. So what we're trying to do here is define what we support (and what we don't) so we can make calls on what we work on (and what we don't work on) should an issue arise.

I guess from our experience we can say that Clang works just as well as gcc, so there must be a very high level of compatibility (and C99 compliance), certainly no non-standard quirks like say MSVC.

If a clang build through up a bug, I would be tempted to look into it - cross compiling the same code is great way to find bugs in C.

Would it perhaps be better if we just say "C99"? Or should we add clang to the list?  Hmm, there might also be other compilers that work :thinking: So I'm tempted to say "C99, and we use gcc for development and testing".

Reply:
Yeah, I think just mentioning C99 will work. 

(BTW on the freedv-gui side, I want to say that C++14 is mandatory but I'm not 100% sure. Regardless, Clang and GCC both compile it fine.)